### ISSUE TITLE
Contributing guidlines

### ISSUE THREAD

**Issue by danielzfranklin** [linux-surface/linux-surface#24]
This project is awesome, and I depend on it. I want to know if it's possible for me to contribute as someone without much experience. It seems like this project doesn't organize in-progress development around the Github issue tracker. Is there somewhere else where that happens? Is there are a place where I could find issues that I could work on?

Thanks,
Daniel

**Comment by chrismin13** [comment]
+1! I actually wanted to write an email to qzed proposing moving all the related repositories from his github user page to a dedicated organization to make it easier to view and manage, so I'm very happy to see this happened.

The bigger problem here that's preventing us from having more people contribute is that there are just too many places discussing Surface Linux compatibility and doing development:
 - The old Linux-surface repo from jakeday which is no longer maintained
 - The r/SurfaceLinux subreddit with its awfully outdated top post.
 - The Kernel repo that was moved to this organization
 - This repo's and the Kernel's wiki
 - Various websites and threads online talking about this effort

I think we should start cutting down on misinformation and multiple sources. I propose we should make a central location for all discussion on Surface Linux where newcomers can read wikis and other people's experiences to set up Linux without being confused by multiple repositories and sources.

However, I understand that hosting a website and a domain requires money of course that as an open source project doesn't exist. So I've been thinking about alternative ways of going about this.

One option was Atlassian's various offerings for development tools. Normally they have a hefty price tag but Open Source projects can apply for an Open Source license which lets you use any products you want for free. And you get the benefits of great integration with Atlassian's products, such as Bitbucket (for Git code hosting), Jira (for issue tracking) and Confluence (their wiki). However, after trying it out, it seems a bit too complicated for our needs as it's mostly geared towards businesses and not open source projects and might a be a bit daunting for new comers. Still up for consideration though and I'd love to hear your opinions.

Another option would be to more actively moderate the Subreddit. This is a great option as we could create a wiki and have discussion over there, while only using Github as an issue tracker and a place for all things related to the code of the project. This would cut down on the discussion taking place right now on the issues section of the github repos by newcomers that can't get support elsewhere. However, this would mean finding the moderators of the r/SurfaceLinux subreddit to add this organization's users as moderators so everyone could contribute. If you want, I could look into that myself and figure out a way to update the Subreddit.

Lastly, there's always the option of creating a webpage, where we could have Forums and a Wiki, much like most other open source projects do (OpenWrt and SpigotMC come to mind). I'll try to contribute as much as possible, but this is a monumental task so I'm not sure how much interest there would be.

Anyways, that's my opinions. I'm always open to discussion and would love to hear your thoughts. I think this is a great project and the work being made shouldn't go to waste. Figuring this out would be a great way to get more people interested in the project and have more people test and contribute.

**Comment by archseer** [comment]
Yeah so qzed initially turned off the issue tracker because we were hoping to keep upstreaming changes to jakeday/linux-surface, so he wanted to keep all the issues in one place. But since we've quite diverged at this point, we decided to go for it and set up a new organization (that way we at least increase our bus factor a bit). So I think the plan is to use this repo's issue tracker as the main place to organize development, once we have all the repos organized.

I've also just messaged the last active r/surfacelinux moderator in hopes that we can get those old threads updated so that newbies aren't installing old kernels that won't work for newer devices.

I think the current plan is to make the wiki, and specifically the install guide (https://github.com/linux-surface/linux-surface/wiki/Installation-and-Setup) the definitive place for guides and device specific quirks. A big part of that is documenting all the common questions that pop up ("why isn't touchscreen working?" "doesn't work on 5.4+ and/or gen7", "why doesn't battery status work?" "you're missing the acpi module", etc.)

**Comment by chrismin13** [comment]
That's great to hear! Thank you for taking the time to do this, it sounds like a good plan for now. Please let us know if you get a reply on Reddit!

**Comment by danielzfranklin** [comment]
Jakeday's kernel has a bunch of still relevant issues that I presume he isn't going to fix in the near future. Would it be acceptable to manually move one to this repo if I've experienced it using the latest in this repo? (Specifically https://github.com/jakeday/linux-surface/issues/395)

**Comment by archseer** [comment]
Yeah I would say reopen anything still relevant in this repo, that way we know it's still not fixed. Maybe backlink to the old issue as well.
